client/components/site-selector/index.jsx

```diff
@@ -222,8 +224,37 @@ export default React.createClass( {
 
 			const isSelected = this.isSelected( site );
 
-			if ( isSelected && this.props.hideSelected ) {
-				return;
+			return (
+				<Site
+					site={ site }
+					href={ siteHref }
+					key={ 'site-' + site.ID }
+					indicator={ this.props.indicator }
+					onSelect={ this.onSiteSelect.bind( this, site.slug ) }
```

Comment:
I'm not sure I like the signature ordering, but we can get around the bind if we update my-site/site component's onSelect to pass through the slug
```javascript
	onSelect: function( event ) {
		if ( this.props.homeLink ) {
			return;
		}

		this.props.onSelect( this.props.site.slug, event );
		event.preventDefault(); // this doesn't actually do anything...
	}
```

And then back in this file pass through
```javascript
onSelect={ this.onSiteSelect }
```